src/protean/core/entity.py

```diff
@@ -458,6 +461,9 @@ def create(cls, *args, **kwargs) -> 'Entity':
             # Set Entity status to saved
             entity.state_.mark_saved()
 
+            # Perform Post-Save Actions
+            entity.post_save()
```

Comment:
This should be implemented using signals, we should not go with this approach. Please take a look at this package https://pythonhosted.org/blinker/, which is also used by flask

Reply:
I disagree. Let's reserve signals for the more powerful `event` and `subscription` mechanism, than entity hooks. Entity hooks ensure all code related to persistence is present in the same place, whereas signals can be spread anywhere in the code and can be disastrous in the wrong hands. It's just too easy to forget signals exist if they are in a different part of the project.

* https://lincolnloop.com/blog/django-anti-patterns-signals/
* https://www.quora.com/Are-Django-signals-an-anti-pattern

Reply:
This implementation is closer (albeit a very simplistic implementation) to `Callbacks` implemented in Rails ActiveRecords: https://api.rubyonrails.org/classes/ActiveRecord/Callbacks.html